### PR TITLE
Add basic benchmark for query pushdowns

### PR DESCRIPTION
**Pull request by noisersup**
# Description

This one was created for the blogpost. As agreed it doesn't run on any external data and will be modified in the future.
As I'm not sure if it met requirements, @AlekSi PTAL.

## Readiness checklist

* [ ] I added unit tests for new functionality or bugfixes.
* [x] I added integration tests for new functionality or bugfixes.
* [ ] I added compatibility tests for new functionality or bugfixes.
* [ ] I made spot refactoring.
* [ ] I updated the documentation.
* [x] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by AlekSi — CHANGES_REQUESTED** [review]
Benchmark _has_ to use `b.N`: https://pkg.go.dev/testing#hdr-Benchmarks

File: integration/benchmarks_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+func BenchmarkPushdowns(b *testing.B) {
+	ctx, coll := setup.Setup(b, shareddata.AllProviders()...)
+
+	res, err := coll.InsertOne(ctx, bson.D{{}})
+	require.NoError(b, err)
+
+	id := res.InsertedID
+
+	// run benchmark with query pushdown
+	b.Run("ObjectID", func(b *testing.B) {
```
Comment (chilagrow) [inline integration/benchmarks_test.go:36]:
```suggestion
	b.Run("Pushdown", func(b *testing.B) {
```

Comment (noisersup) [reply integration/benchmarks_test.go:36]: I would suggest to leave it as it is. In the future we would implement pushdowns for other types/conditions and it maybe could be useful to benchmark how they affect on our performance. WDYT?

Comment (chilagrow) [reply integration/benchmarks_test.go:36]: ok :hugs:

**Review by chilagrow — DISMISSED** [review]

File: integration/benchmarks_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+			var res []bson.D
+			err = cur.All(ctx, &res)
+			require.NoError(b, err)
+		}
+	})
+
+	// run benchmark without query pushdown
+	b.Run("NoPushdown", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			cur, err := coll.Find(ctx, bson.D{{"_id", "string"}})
```
Comment (rumyantseva) [inline integration/benchmarks_test.go:50]: Well, after the `settings` refactoring task, this one (stringified ids) has query pushdown too 😅 
I'd suggest a non-`_id` field for now (maybe `v` then), but as soon as we start supporting more query pushdowns, that check could become outdated.

**Review by rumyantseva — CHANGES_REQUESTED** [review]
Currently, `NoPushdown` does a pushdown :)

File: integration/benchmarks_test.go (hunk 2 above)
Comment (noisersup) [reply integration/benchmarks_test.go:50]: Cool! Then at least we don't need to insert the objectid document

Comment (noisersup) [reply integration/benchmarks_test.go:53]: Great! I've added string ids as a seperate subtest for now. I've also added NotEmpty checks to make sure that we get the data for the every test

**Review by rumyantseva — APPROVED** [review]
LGTM

File: integration/benchmarks_test.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+func BenchmarkPushdowns(b *testing.B) {
+	ctx, coll := setup.Setup(b, shareddata.AllProviders()...)
+
+	res, err := coll.InsertOne(ctx, bson.D{{}})
+	require.NoError(b, err)
+
+	id := res.InsertedID
+
+	_, err = coll.InsertOne(ctx, bson.D{{"v", "foo"}})
+	require.NoError(b, err)
```
Comment (AlekSi) [inline integration/benchmarks_test.go:36]: Seems unused?

**Review by chilagrow — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
LGTM

File: integration/benchmarks_test.go (hunk 3 above)
Comment (noisersup) [reply integration/benchmarks_test.go]: fixed

**Review by AlekSi — APPROVED** [review]

**Pull request merged**